pkg/chunk/repair.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package chunk
```

Comment:
also - I have a bit of a problem with having a `chunk` package per-se. The `Chunk` type is in `swarm` package, so having a `chunk` package that does not even relate to the actual chunk data type seems counter intuitive for me. `recovery` would be a possible alternative

Reply:
my actual intention of adding a chunk package is to move the chunks related things from swarm to here. This was discussed here as part of trojan chunks  https://github.com/ethersphere/bee/pull/429/files#r453443685

Reply:
i don't think that the chunk repair business logic should be in the same package as the chunk type.

Reply:
> afair, thought u had decided to use `prod/client prod/server`.

same here.

asking for content repair and globally pinning it are 2 related but separate things.

if you we are not going to separate them now, please create an issue to so that this is documented.

(if this issue already exists, i apologize, but i have not been able to find it)

Reply:
> asking for content repair and globally pinning it are 2 related but separate things.

@mortelli I dont understand. They are same, one side is asking (client) other side is doing (server). Both are what is called chunk repair in toto. I never understood why it was called prod in the first place. so changed it to retrieve per @acud's comment.

Reply:
ultimately i don't mind one name or the other, but you've already differentiated them by calling them client and server:

- all nodes should be able to request a chunk repair (this is default behaviour)
- only nodes that are configured as such will be able to do the repair routine (global pinners)

so `prod/client` and `prod/server` (replace `prod` with `recovery` if you wish) would keep these things related but separate.

Reply:
so you want to block on this?